.github/workflows/dev_deploy.yml

```diff
@@ -0,0 +1,165 @@
+---
+name: Deploy/Destroy Branch Dev Environment
+
+on:
+  pull_request:
+    types: [labeled, closed]
+    
+env:
+  TERRAFORM_CLOUD_TOKENS: app.terraform.io=${{ secrets.HUSHLINE_DEV_TF_TOKEN }}
+  GITHUB_TOKEN: ${{ github.token }}
+  WORKSPACE_NAME: hushline-dev-${{ github.head_ref }}
+  TF_PROJECT_HUSH_LINE_DEV: prj-iEruEQFmaNTCRAtA
+  DEV_TF_PATH: terraform/dev
+  DO_APP_NAME: hushline-dev-${{ github.head_ref }}
+  HUSHLINE_INFRA_REPO: scidsg/hushline-infra
+  HUSHLINE_INFRA_REF: dev-team
+
+jobs:
+  deploy:
+    if: ${{ github.event.action == 'labeled' && github.event.label.name == 'deploy' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout terraform
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.HUSHLINE_INFRA_REPO }}
+          ref: ${{ env.HUSHLINE_INFRA_REF }}
+          token: ${{ secrets.HUSHLINE_INFRA_TOKEN }}
+
+      - name: Create workspace
+        run: |
+          set -e
```

Comment:
Of all the tools I found to create tf workspaces, none of them seemed to be able to set the workspace "working-directory" or specify the project to create workspace in. This second part is critical because the TF token used by this workflow only has permissions inside of the dev project.